src/upload-helper.ts

```diff
@@ -145,3 +145,34 @@ export class UploadHelper {
     return await pMap(filesList, uploader, { concurrency });
   }
 }
+
+/**
+ * expandGlob compiles the list of all files in the given directory for
+ * the provided glob.
+ *
+ * @param directoryPath The path to the directory.
+ * @param glob Glob pattern to use for searching. If the empty string, a
+ * match-all pattern is used instead.
+ * @return Sorted list of files in posix form.
+ */
+export async function expandGlob(
+  directoryPath: string,
+  glob: string,
+): Promise<string[]> {
+  const pth = toPosixPath(path.posix.join(directoryPath, glob));
+  const filesList = await globby([pth], {
+    dot: true,
```

Comment:
Should we make this an option defaulting to false to maintain backwards compat?

Reply:
I find it surprising that we don't match what gsutil and gcloud do here, so I'm thinking of this as a "bug" more than a feature.

Reply:
I wasnt aware thats default gsutil behavior so this sgtm. Maybe in the changelog we can include how to keep existing behavior via ignorefile?